### DISCUSSION TITLE
How to reverse port biding (Phusion Passenger and Fastify)?

### DISCUSSION THREAD

**Discussion by rubenferreira97** [fastify/fastify#4240] [Q&A] [answered]
I am trying to setup a node.js app on CPanel using Phusion Passenger.
After some research I found that Phusion Passenger reverses control on the port binding.
https://www.phusionpassenger.com/library/indepth/nodejs/reverse_port_binding.html

There are solutions for:

- node-http-proxy
- Express.js
- Hapi.js

**Express.js example:**
```
if (typeof(PhusionPassenger) !== 'undefined') {
    PhusionPassenger.configure({ autoInstall: false });
}

var express = require('express');
var app = express();
app.get('/', function(req, res) {
    var body = 'Hello World';
    res.setHeader('Content-Type', 'text/plain');
    res.setHeader('Content-Length', body.length);
    res.end(body);
});

if (typeof(PhusionPassenger) !== 'undefined') {
    app.listen('passenger');
} else {
    app.listen(3000);
}
```

Is it possible to setup this use case in fastify? I have look to `FastifyListenOptions` but don't know which property should I try out.

**Comment by mcollina** [comment]
I think your best option is to contact Phusion Passenger support.

**Comment by Eomm** [comment, marked as answer]
Duplicated
https://stackoverflow.com/questions/73563025/how-to-reverse-port-biding-phusion-passenger-and-fastify
